pysph/sph/wc/shift.py

```diff
@@ -0,0 +1,226 @@
+from math import sqrt
+from compyle.api import declare
+from pysph.sph.equation import Equation
+from pysph.base.reduce_array import serial_reduce_array
+from pysph.solver.tools import Tool
+
+
+class SimpleShift(Equation):
+    """See the paper by R. Xu et al. (JCP 2009), equation(35)
+    """
+    def __init__(self, dest, sources, const=0.04):
+        self.beta = const
+        super(SimpleShift, self).__init__(dest, sources)
+
+    def py_initialize(self, dst, t, dt):
+        from math import sqrt
+        vmag = dst.u[:]*dst.u[:] + dst.v[:]*dst.v[:] + dst.w[:]*dst.w[:]
+        vmax = serial_reduce_array(vmag, 'max')
+        dst.vmax[0] = sqrt(vmax)
+
+    def loop_all(self, d_idx, d_x, d_y, d_z, s_x, s_y, s_z, s_u, s_v, s_w,
+                 d_u, d_v, d_w, d_vmax, d_dpos, dt, N_NBRS, NBRS):
+        i, s_idx = declare('int', 2)
+        ri = 0.0
+        dxi = 0.0
+        dyi = 0.0
+        dzi = 0.0
+        eps = 1.0e-08
+        for i in range(N_NBRS):
+            s_idx = NBRS[i]
+            xij = d_x[d_idx] - s_x[s_idx]
+            yij = d_y[d_idx] - s_y[s_idx]
+            zij = d_z[d_idx] - s_z[s_idx]
+            rij = sqrt(xij*xij + yij*yij + zij*zij)
+            r3ij = rij * rij * rij
+            dxi += xij / (r3ij + eps)
+            dyi += yij / (r3ij + eps)
+            dzi += zij / (r3ij + eps)
+            ri += rij
+
+        ri = ri/N_NBRS
+        fac = self.beta * ri*ri * d_vmax[0] * dt
+        d_dpos[d_idx*3] = fac*dxi
+        d_dpos[d_idx*3 + 1] = fac*dyi
+        d_dpos[d_idx*3 + 2] = fac*dzi
+
+        d_x[d_idx] += d_dpos[d_idx*3]
+        d_y[d_idx] += d_dpos[d_idx*3 + 1]
+        d_z[d_idx] += d_dpos[d_idx*3 + 2]
+
+
+class FickianShift(Equation):
+    """See the paper by S. J. Lind et al. (JCP 2012), equation(21-24),
+    for the constant see A. Skillen et al. (CMAME 2013), equation(13).
+    """
+    def __init__(self, dest, sources, const=10, tensile_const=0.2,
```

Comment:
Why the name `const`, can there be a better name that is similar to the one used in the paper?

Reply:
the papers names it as 'C', now changed it to 'fickian_const', should I change it ?